File: medium/3169. Count Days Without Meetings/main.py
from typing import List


class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        meetings.sort()

        merged = []

        for meet in meetings:
            if merged and merged[-1][1] >= meet[0]:
                merged[-1][1] = max(merged[-1][1], meet[1])
            else:
                merged.append(meet)

        result = 0
        for meet in merged:
            result += meet[1] - meet[0] + 1

        return days - result

File: medium/3169. Count Days Without Meetings/main.go
package main

import "sort"

func countDays(days int, meetings [][]int) int {
	sort.Slice(meetings, func(i, j int) bool {
		return meetings[i][0] < meetings[j][0]
	})

	merged := make([][]int, 0, len(meetings))
	for _, meet := range meetings {
		if len(merged) == 0 || meet[0] > merged[len(merged)-1][1] {
			merged = append(merged, meet)
		} else {
			merged[len(merged)-1][1] = max(merged[len(merged)-1][1], meet[1])
		}
	}

	var result int
	for _, meet := range merged {
		result += meet[1] - meet[0] + 1
	}

	return days - result
}
